src/analysis/processing/qgsalgorithmrasterrank.cpp

```diff
@@ -0,0 +1,309 @@
+/***************************************************************************
+                         qgsalgorithmrasterrank.cpp
+                         ---------------------
+    begin                : February 2024
+    copyright            : (C) 2024 by Mathieu Pellerin
+    email                : mathieu at opengis dot ch
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmrasterrank.h"
+#include "qgsrasterfilewriter.h"
+#include "qgsrasterprojector.h"
+
+///@cond PRIVATE
+
+QString QgsRasterRankAlgorithm::name() const
+{
+  return QStringLiteral( "rasterrank" );
+}
+
+QString QgsRasterRankAlgorithm::displayName() const
+{
+  return QObject::tr( "Raster rank" );
+}
+
+QStringList QgsRasterRankAlgorithm::tags() const
+{
+  return QObject::tr( "raster,rank" ).split( ',' );
+}
+
+QString QgsRasterRankAlgorithm::group() const
+{
+  return QObject::tr( "Raster analysis" );
+}
+
+QString QgsRasterRankAlgorithm::groupId() const
+{
+  return QStringLiteral( "rasteranalysis" );
+}
+
+QString QgsRasterRankAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Performing a cell-by-cell analysis in which output values match the rank of a sorted list of overlapping cell values from input layers." );
```

Comment:
Could you expand on these? This algorithm isn't immediately clear, so I think more detail would help users

Reply:
Definitely needs documenting that the output is multi-band when multiple ranks are entered, that wasn't obvious to me until reading the code!